Add perf tests for Acquire Token calls with serialization cache

pmaytak opened the pull request:
Fixes # 
<!-- Add the issue number above. If this PR only partially fixes the issue, remove the Fixes word above so that the issue is not automatically closed. -->

**Changes proposed in this request**
<!-- Concisely list the changes. -->
<!-- If helpful, describe how and why the bug was fixed. -->
<!-- If needed, describe how to review (ex. which files have the primary changes, which are nit changes, etc.) -->

**Testing**
<!-- Have unit, integration, etc. tests been added? Describe any relevant testing that has been done. -->
<!-- Mention if any and what extra manual testing is needed during the release. -->

**Performance impact**
<!-- Describe any applicable performance impact or performance testing done. -->

bgavrilMS commented:
For S2S: 

I don't know if it's realistic to have 1000 tokens per partition. Maybe we should keep the number of resources at 10, and vary the number of tenants: 1k, 10k, 100k, 1M ? The (1M, 10) would be interesting to see. 

For OBO: same comment.

bgavrilMS commented on tests/Microsoft.Identity.Test.Performance/Helpers/InMemoryCache.cs:
```diff
@@ -0,0 +1,52 @@
+{
+    public class InMemoryCache
+    {
+        private readonly ConcurrentDictionary<string, byte[]> _memoryCache = new ConcurrentDictionary<string, byte[]>();
```
It would be better to use the MemoryCache from Id.Web here, maybe with an eviction policy.

bgavrilMS commented on tests/Microsoft.Identity.Test.Performance/Helpers/InMemoryCache.cs:
```diff
@@ -0,0 +1,52 @@
+
+namespace Microsoft.Identity.Test.Performance.Helpers
+{
+    public class InMemoryCache
```
It would be better to use the MemoryCache from Id.Web here, maybe with an eviction policy.

bgavrilMS reviewed (changes requested):
1. We should use the memory cache from Id.Web for comparison, as that is the one we recomend. The Id.Web layer does extra work, such as setting policies etc. 
2. Not sure if a lower number of partitions is relevant. We should use a high number of partitions and a lower number of tokens per partition. These are the real life scenarios IMO.

pmaytak commented:
> 1. We should use the memory cache from Id.Web for comparison, as that is the one we recomend. The Id.Web layer does extra work, such as setting policies etc.
> 2. Not sure if a lower number of partitions is relevant. We should use a high number of partitions and a lower number of tokens per partition. These are the real life scenarios IMO.

- Added results above for Identity Web cache. Compared to in-memory cache, Id.Web cache is usually a bit slower, but not significantly.
- Added results for test cases with multiple partitions and few tokens per partition.
- Added test case of 1M partitions, but there were issues running the tests, probably due to memory constraints. As expected though, the perf is similar to other tests - number of partitions don't affect perf too much.
- Yes, high number of tokens per partition is probably not common / realistic, but I think it can give good perspective on perf of internal methods, like filtering.
